File: Leetcode/python3/medium/1689_Partitioning Into Minimum Number Of Deci-Binary Numbers.py
#
# @lc app=leetcode id=1689 lang=python3
#
# [1689] Partitioning Into Minimum Number Of Deci-Binary Numbers
#


# @lcpr-template-start
from preImport import *
# @lcpr-template-end
# @lc code=start
class Solution:
    def minPartitions(self, n: str) -> int:
        return ord(max(n)) - ord('0')
# @lc code=end

File: Leetcode/rust/medium/1689_Partitioning Into Minimum Number Of Deci-Binary Numbers.rs
/*
 * @lc app=leetcode id=1689 lang=rust
 *
 * [1689] Partitioning Into Minimum Number Of Deci-Binary Numbers
 */

// @lc code=start
use std::cmp::max;

impl Solution {
    pub fn min_partitions(n: String) -> i32 {
        let mut ans = 0;
        for c in n.chars() {
            ans = max(ans, c as u8 - b'0');
        }
        ans as i32
    }
}
// @lc code=end

